src/Design.cc

```diff
@@ -98,6 +98,120 @@ void Design::readDef(const std::string& file_name,
                child);
 }
 
+sta::Network* Design::cmdLinkedNetwork()
+{
+  sta::Network *network = sta::Sta::sta()->cmdNetwork();;
+  if (network->isLinked()) {
+    return network;
+  }
+
+  getLogger()->error(utl::ORD, 104, "STA network is not linked.");
+  return nullptr;
+
+}
+
+sta::Graph* Design::cmdGraph()
+{
+  cmdLinkedNetwork();
+  return sta::Sta::sta()->ensureGraph();
+}
+
+sta::Vertex** Design::vertices(const sta::Pin *pin)
+{
+  sta::Vertex *vertex, *vertex_bidirect_drvr;
+  static sta::Vertex *vertices[2];
+
+  cmdGraph()->pinVertices(pin, vertex, vertex_bidirect_drvr);
+  vertices[0] = vertex;
+  vertices[1] = vertex_bidirect_drvr;
+  return vertices;
+}
+
+std::vector<float> Design::arrivalsClk(const sta::RiseFall *rf,
+	                                      sta::Clock *clk,
+	                                      const sta::RiseFall *clk_rf,
+                                        sta::Vertex *vertex)
+{
+  sta::Sta *sta = sta::Sta::sta();
+  std::vector<float> arrivals;
+  const sta::ClockEdge *clk_edge = nullptr;
+
+  if (clk){
+    clk_edge = clk->edge(clk_rf);
+  }
+
+  for (auto path_ap : sta->corners()->pathAnalysisPts()) {
+    arrivals.push_back(sta::delayAsFloat(sta->vertexArrival(vertex, rf, clk_edge, path_ap, nullptr)));
+  }
+  return arrivals;
+}
+
+std::string Design::getITermName (odb::dbITerm* ITerm)
+{
+    auto MTerm_name = ITerm->getMTerm()->getName();
+    auto inst_name  = ITerm->getInst()->getName();
+    return inst_name + "/" + MTerm_name;
+}
+
+bool Design::isTimeInf(float time) {
+  return (time > 1e+10 || time < -1e+10); 
+}
+
+float Design::getPinArrivalTime(sta::Clock *clk,
+                               const sta::RiseFall *clk_rf,
+                               sta::Vertex *vertex,
+                               const std::string& arrrive_or_hold)
```

Comment:
Why is this a string and not sta::RiseFall ?

Reply:
I believe this is done this way because because we pass in a string object from the python interface!

Reply:
If you really want to use a string then you should provide a typemap in the .i but keep the c++ interface strongly typed.